### PR TITLE
[Enhancement] Cookies setup for JWT

### LINKED ISSUE #46: Set token in cookies for authentication.

### PR DESCRIPTION
**Pull request by ViralRuparel**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #46 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

### Checklist for opening a PR
Please, go through these steps before you open a PR.
- [x] I have created an issue for this PR.
- [x] I have properly named the Pull request with **[tag]** before the name of PR.
- [x] **I am choosing the correct base branch to merge**. 

### Checklist for submitting your PR
Please, go through these steps before you resolve you **[WIP]** status.
- [x] I have implemented the functions and relevant test cases.
- [x] All my test cases are passing.
- [x] I have made corresponding changes to the documentation
- [x] I have rebased my branch.
- [x] I have added atleast 2 reviewers 
- [x] I have provided a description of changes.
- [x] I have attached relevant screenshots/gifs ( if UI has changed.)
- [x] I have only one commit. (if not, squash them into one commit).
- [x] I have added `closes #XXXX` in my comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Comment by ViralRuparel** [comment]
closes #46

File: server/src/constants/token.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import dotenv from "dotenv";
 dotenv.config();
 export default {
-  JWT_KEY: process.env.JWT_KEY
+  JWT_KEY: process.env.JWT_KEY,
+  TIMEOUT: 86400000
```
Comment (rishabh1403) [inline server/src/constants/token.js]: shoul dbe a constant , and go into constants file

File: server/src/middleware/auth.js [hunk 2]
```diff
@@ -3,15 +3,19 @@ import User from "../models/user";
...
-    const user = await User.findOne({ _id: data._id, "tokens.token": token });
+    if (!token) {
+      return res
+        .status(401)
+        .send({ error: "Not authorized to access this resource" });
+    }
+    const data = await jwt.verify(token, constants.JWT_KEY);
+    const user = await User.findOne({ _id: data._id });
     if (!user) {
       throw new Error();
```
Comment (rishabh1403) [inline server/src/middleware/auth.js]: have message in error, else you'll not know if your custom error is thrown or something else

**Review by rishabh1403 — APPROVED** [review]
Minor changes then merge

File: server/src/middleware/auth.js (hunk 2 above)
Comment (ViralRuparel) [reply server/src/middleware/auth.js]: I will add it to booking-api's PR.

**Review by vineetvk01 — APPROVED** [review]
lgtm

**Pull request merged**